Apps/Src/UpdateVelocity.c

```diff
@@ -22,13 +22,52 @@ void Task_UpdateVelocity(void* p_arg){
         //Read global pedal percentage
         uint8_t pedalPercentage = car_state->AccelPedalPercent;
 
-        if(car_state->CruiseControlEnable && car_state -> CruiseControlSet){
-            car_state->DesiredVelocity = car_state->CruiseControlVelocity;
-            car_state->DesiredMotorCurrent = 1.0f;
-        }else{
+        if(car_state->CREnable == CRUISE){
+            if(car_state->IsRegenBrakingAllowed){
+                if(car_state -> CruiseControlSet){
+                    car_state->DesiredVelocity = car_state->CurrentVelocity;
```

Comment:
CruiseControlSet should be on whenever the car is in cruise control with a velocity already set. This line will keep setting the desired velocity to the current velocity when it should be set to CruiseControlVelocity. The cruise control velocity needs to be set seperately whenever the cruise set button is pressed.